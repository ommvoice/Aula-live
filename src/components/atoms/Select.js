import React from 'react';

const Select = ({ defaultValue, onChange, options }) => {
	return (
		<div className="col-6">
			<select defaultValue={defaultValue} onChange={onChange} className="form-control margin-btm">
				{
					options && options.map((option) => (
							<option key={option.value} value={option.value}>{option.title}</option>
					))
				}
			</select>
		</div>
	);
}

export default Select;
