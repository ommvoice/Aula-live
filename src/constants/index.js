export const API = 'https://rss.itunes.apple.com';
export const PROXY = 'https://cors-anywhere.herokuapp.com/';
export const SONG_CATEGORIES = [
  { title: 'Top Albums', value: 'top-albums' },
	{ title: 'Top Songs', value: 'top-songs' },
	{ title: 'Hot Tracks', value: 'hot-tracks' },
	{ title: 'New Releases', value: 'new-releases' },
	{ title: 'Coming Soon', value: 'coming-soon' }
];
