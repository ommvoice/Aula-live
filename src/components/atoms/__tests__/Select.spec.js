import React from 'react';
import { shallow } from 'enzyme';
import Select from '../Select';
import { SONG_CATEGORIES } from '../../../constants';

describe('Select', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Select options={SONG_CATEGORIES} />);

    expect(wrapper).toMatchSnapshot();
  });
});
