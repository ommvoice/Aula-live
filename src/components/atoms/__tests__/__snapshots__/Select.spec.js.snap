// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="col-6"
>
  <select
    className="form-control margin-btm"
  >
    <option
      key="top-albums"
      value="top-albums"
    >
      Top Albums
    </option>
    <option
      key="top-songs"
      value="top-songs"
    >
      Top Songs
    </option>
    <option
      key="hot-tracks"
      value="hot-tracks"
    >
      Hot Tracks
    </option>
    <option
      key="new-releases"
      value="new-releases"
    >
      New Releases
    </option>
    <option
      key="coming-soon"
      value="coming-soon"
    >
      Coming Soon
    </option>
  </select>
</div>
`;
